packages/advisor-components/src/RuleDetails/RuleDetails.tsx

```diff
@@ -0,0 +1,247 @@
+import { Likelihood } from '../types';
+import { TopicRhel } from '../types';
+import { Rating } from '../types';
+import { RuleContentOcp } from '../types';
+import { RuleContentRhel } from '../types';
+import { AdvisorProduct } from '../types';
+import { string } from 'prop-types';
+
+interface RuleDetailsBaseProps {
+  rule: RuleContentOcp | RuleContentRhel;
+  resolutionRisk?: number;
+  resolutionRiskDesc?: string;
+  header?: React.ReactNode;
+  isDetailsPage: boolean;
+  topics: TopicRhel[];
+  /**
+   * onVoteClick - a callback used to update the rating of a particular rule
+   * @param {string} ruleId - ID (usually in plugin|error_key format) of the rule that needs to be updated
+   * @param {number} newRating rating (-1, 0, 1)
+   */
+  onVoteClick?: (ruleId: string, calculatedRating: Rating) => unknown;
+  onViewAffectedClick?: (ruleId: string) => unknown;
+}
+
+interface RuleDetailsProps extends RuleDetailsBaseProps, ContextWrapperProps {}
+
+const RuleDetailsBase: React.FC<RuleDetailsBaseProps> = (props) => {
+  const intl = useIntl();
+  const product = useContext(ProductContext);
+  const { header, rule, isDetailsPage, topics, onVoteClick, onViewAffectedClick, resolutionRisk, resolutionRiskDesc, children } = props;
+
+  const renderDescription = () => {
+    const ruleDescription = (data: string, isGeneric = false) =>
+      typeof data === 'string' &&
+      Boolean(data) && (
+        <span className={isGeneric ? 'genericOverride' : ''}>
+          <Markdown rehypePlugins={[rehypeRaw, rehypeSanitize]}>{data}</Markdown>
+        </span>
+      );
+
+    switch (product) {
+      case AdvisorProduct.ocp:
+        return ruleDescription(rule.generic, true);
+        break;
+      case AdvisorProduct.rhel:
+        return isDetailsPage ? ruleDescription(rule.generic, true) : ruleDescription(rule.summary);
+    }
+  };
+
+  const renderViewAffected = () => {
+    switch (product) {
+      case AdvisorProduct.ocp:
+        if ((rule as RuleContentOcp).impacted_clusters_count > 0) {
+          return (
+            <StackItem>
+              <Link key={`${rule.rule_id}-link`} onClick={() => onViewAffectedClick && onViewAffectedClick(rule.rule_id)} to="#">
+                {intl.formatMessage(messages.viewAffectedClusters, {
+                  clusters: (rule as RuleContentOcp).impacted_clusters_count,
+                })}
+              </Link>
+            </StackItem>
+          );
+        }
+        break;
+      case AdvisorProduct.rhel:
+        if ((rule as RuleContentRhel).impacted_systems_count > 0) {
+          return (
+            <StackItem>
+              <Link key={`${rule.rule_id}-link`} onClick={() => onViewAffectedClick && onViewAffectedClick(rule.rule_id)} to="#">
+                {intl.formatMessage(messages.viewAffectedSystems, {
+                  systems: (rule as RuleContentRhel).impacted_systems_count,
+                })}
+              </Link>
+            </StackItem>
+          );
+        }
+    }
```

Comment:
Same as above, not a huge fan of switches. Any reason to use `switch`?

Reply:
Well, the thing is, once a new product supporting Advisor will come with its special rule structure and so on, the switch will provide a better syntax to treat multiple values of AdvisorProduct. However, currently, this is not an issue, I think you point is legit.. reworked it in the previous commits